### PR TITLE
Add and edit docs for generate subcommands 

### PR DESCRIPTION
**Pull request by jbogarthyde**
WIP - transfer edited doc content for generation schematics from draft to source files

### CODE REVIEW

File: packages/schematics/angular/guard/schema.json [hunk 1]
```diff
@@ -15,18 +16,18 @@
     },
     "spec": {
       "type": "boolean",
-      "description": "Specifies if a spec file is generated.",
+      "description": "When true (the default), generates a  \"spec.ts\" test file for the new guard.",
```
Comment (alan-agius4) [inline packages/schematics/angular/guard/schema.json]: Nit: would be nice to have the same descriptions for the same flag. 

For instance sometimes, it is `"When true, generates...`` and sometimes it is `When true (the default), generates ..`

**Comment by alan-agius4** [comment]
@jbogarthyde, Can you please rebase?

File: packages/schematics/angular/application/schema.json [hunk 2]
```diff
@@ -3,14 +3,15 @@
   "id": "SchematicsAngularApp",
   "title": "Angular Application Options Schema",
   "type": "object",
+  "description": "Generates a new basic app definition in the \"projects\" subfolder of the workspace.",
   "properties": {
     "projectRoot": {
-      "description": "The root directory of the new application.",
+      "description": "The root folder of the new app.",
```
Comment (alan-agius4) [inline packages/schematics/angular/application/schema.json]: just a small nit, but in general we use `directory` instead of `folder` as terminology

Comment (jbogarthyde) [reply packages/schematics/angular/application/schema.json]: Jury is out on this one. I tend to prefer folder. 
@jenniferfell 
The Author's Style Guide does not address the issue -- and in fact, that page uses "folder" 19 times and "directory" 12 times - sometimes in the same sentence.

**Comment by jbogarthyde** [comment]
Moved these commits to docs-preview branch - PR Docs preview #12837

**Pull request closed without merging**